### PR TITLE
feat: add support for markdown.nvim

### PR DESCRIPTION
**Pull request by sjclayton**
Add support for [markdown.nvim](https://github.com/MeanderingProgrammer/markdown.nvim) for improved Markdown rendering in Neovim.

Example of plugin in action:

### CODE REVIEW

**Comment by sjclayton** [comment]
Additional sample to show bullets and an ordered list:

**Comment by sjclayton** [comment]
Current state after adjustments in latest commit:

File: lua/rose-pine.lua [hunk 1]
```diff
@@ -923,6 +923,24 @@ local function set_highlights()
...
+		RenderMarkdownChecked = { fg = palette.foam },
+		RenderMarkdownCode = { bg = palette.overlay },
+		RenderMarkdownCodeInline = { fg = palette.text, bg = palette.overlay },
+		RenderMarkdownDash = { fg = palette.muted },
+		RenderMarkdownH1Bg = { bg = palette.iris, blend = 20 },
+		RenderMarkdownH2Bg = { bg = palette.foam, blend = 20 },
+		RenderMarkdownH3Bg = { bg = palette.rose, blend = 20 },
+		RenderMarkdownH4Bg = { bg = palette.gold, blend = 20 },
+		RenderMarkdownH5Bg = { bg = palette.pine, blend = 20 },
+		RenderMarkdownH6Bg = { bg = palette.leaf, blend = 20 },
```
Comment (mvllow) [inline lua/rose-pine.lua:938]: We should use `groups.<h1...h6>` here. We can also update `groups.h6` to "leaf" if necessary.

**Comment by mvllow** [comment]
Thank you :)

**Pull request merged**